src/Module.h

```diff
@@ -164,6 +167,17 @@ class Module {
     const std::vector<Module> &submodules() const;
     // @}
 
+    /** Tries to locate the offloaded CUDA PTX assembly contained in this Module.
+     * Might return a nullptr in case such buffer is not present in this Module.
+     */
+    const Buffer<> *get_cuda_ptx_assembly_buffer() const;
```

Comment:
Buffers are already a pointer class that can be checked for nullity with buffer.defined(), so it's probably better to return an buffer than a pointer to a buffer.